src/services/legal-services.ts

```diff
@@ -291,24 +308,135 @@ export default class LegalServices {
    * @returns a promise to return the business info
    */
   static async fetchBusinessInfo (businessId: string): Promise<BusinessIF> {
-    const url = `businesses/${businessId}`
+    const url = `${this.legalBusinessUrl}businesses/${businessId}`
 
-    return axios.get(url).then(response => {
-      const data = response?.data
-      if (!data) throw new Error('Invalid API response')
-      return data.business
-    })
+    return axios.get(url, this.config)
+      .then(response => {
+        const data = response?.data
+        if (!data) throw new Error('Invalid API response')
+        return data.business
+      })
   }
 
   /**
    * Fetches the current account's authorized actions (permissions).
    * @returns a promise to return the list of authorized actions
    */
   static async fetchAuthorizedActions (): Promise<AuthorizedActions[]> {
-    return axios.get('permissions').then(response => {
-      const data = response?.data
-      if (!data) throw new Error('Invalid API response')
-      return data.authorizedPermissions
-    })
+    const url = `${this.legalBusinessUrl}permissions`
+
+    return axios.get(url, this.config)
+      .then(response => {
+        const data = response?.data
+        if (!data) throw new Error('Invalid API response')
+        return data.authorizedPermissions
+      })
+  }
+
+  /**
+   * Gets a pre-signed URL for the specified filename.
+   * @param filename the file name
+   * @returns the presigned url object
+   */
+  static async getPresignedUrl (fileName: string): Promise<PresignedUrlIF> {
+    const url = `${this.legalBusinessUrl}documents/${fileName}/signatures`
+
+    return axios.get(url, this.config)
+      .then(response => {
+        const data = response?.data
+        if (!data) {
+          throw new Error('Invalid API response')
+        }
+        return data
+      })
+  }
+
+  /**
+   * Uploads the specified file to the specified (Minio) URL.
+   * @param url the URL to upload to
+   * @param file the file to upload
+   * @param key the file key
+   * @param userId the file user id
+   * @returns a promise to return the axios response or the error response
+   */
+  static async uploadToUrl (url: string, file: File, key: string, userId: string): Promise<AxiosResponse> {
+    // add/override headers
+    const extraConfig = {
+      headers: {
+        'Content-Disposition': `attachment; filename=${file.name}`,
+        'Content-Type': file.type,
+        'x-amz-meta-userid': `${userId}`,
+        'x-amz-meta-key': `${key}`
+      }
+    }
+
+    return axios.put(url, file, merge(this.config, extraConfig))
```

Comment:
The config parameter adds/overrides the default config, but here I merge this file's config with the extra config above -- so basically the final config is 3 pieces put together (default + file config + extra config).